File: Web/RMA-Member-Portal/memberportal-app/src/app/shared/models/user-registration-details.ts
import { BaseClass } from '../../core/models/base-class.model';
import { IdTypeEnum } from '../enums/id-type.enum';
import { PortalTypeEnum } from '../enums/portal-type-enum';
import { UserAddress } from './user-address.model';
import { UserContact } from './user-contact.model';

export class UserRegistrationDetails extends BaseClass {
    userDetailsId: number;
    saId: string;
    passportNo: string;
    name: string;
    surname: string;
    dateOfBirth: Date;
    memberTypeId: number;
    userProfileType: number;
    idTypeEnum: IdTypeEnum;
    companyRegistrationNumber: string;
    healthCareProviderId: number;
    brokerFspNumber: string;
    isInternalUser: boolean;

    userAddress: UserAddress;
    userContact: UserContact;

    password?: string;
    userExistInActivationTable?: boolean;
    userActivationLinkIsActive: boolean;
    userActivationMessage: string;
    userId: number;
    rolePlayerId: number;
    passportExpiryDate?: Date;
    portalType: PortalTypeEnum;
}

File: Service/Admin/RMA.Service.Admin.BusinessProcessManager.Contracts/Entities/UserRegistrationDetails.cs
using RMA.Service.Admin.MasterDataManager.Contracts.Enums;

using System;

namespace RMA.Service.Admin.BusinessProcessManager.Contracts.Entities
{
    public class UserRegistrationDetails
    {
        public int UserDetailsId { get; set; }
        public string SaId { get; set; }
        public string PassportNo { get; set; }
        public string Name { get; set; }
        public string Surname { get; set; }
        public DateTime DateOfBirth { get; set; }
        public UserProfileTypeEnum UserProfileType { get; set; }
        public IdTypeEnum IdTypeEnum { get; set; }
        public string CompanyRegistrationNumber { get; set; }
        public int HealthCareProviderId { get; set; }
        public string BrokerFspNumber { get; set; }

        public UserAddress UserAddress { get; set; }
        public UserContact UserContact { get; set; }
        public bool IsVopdPassed { get; set; }
        public DateTime? PassportExpiryDate { get; set; }
    }
}